src/cellarr/dataloader.py

```diff
@@ -244,56 +254,60 @@ def __init__(
-            self.cell_metadata_tdb.df[:].shape[0],
-            self.gene_annotation_tdb.df[:].shape[0],
+            self.cell_metadata_tdb.shape[0],
+            self.gene_annotation_tdb.shape[0],
         )
 
         # limit to cells with labels
-        query_condition = f"{self.label_column} != 'nan'"
+        query_condition = f"{self.label_column_name} != 'nan'"
         self.data_df = subset_frame(
             self.cell_metadata_tdb,
             query_condition,
-            columns=[self.study_column, self.label_column],
+            columns=[self.study_column_name, self.label_column_name],
         )
 
         # limit to labels that exist in at least 2 studies
         study_celltype_counts = (
-            self.data_df[[self.study_column, self.label_column]]
+            self.data_df[[self.study_column_name, self.label_column_name]]
             .drop_duplicates()
-            .groupby(self.label_column)
+            .groupby(self.label_column_name)
             .size()
             .sort_values(ascending=False)
         )
         well_represented_labels = study_celltype_counts[study_celltype_counts > 1].index
         self.data_df = self.data_df[
-            self.data_df[self.label_column].isin(well_represented_labels)
+            self.data_df[self.label_column_name].isin(well_represented_labels)
         ]
 
         self.val_df = None
         if self.val_studies is not None:
             # split out validation studies
             self.val_df = self.data_df[
-                self.data_df[self.study_column].isin(self.val_studies)
+                self.data_df[self.study_column_name].isin(self.val_studies)
             ]
             self.data_df = self.data_df[
-                ~self.data_df[self.study_column].isin(self.val_studies)
+                ~self.data_df[self.study_column_name].isin(self.val_studies)
             ]
             # limit validation celltypes to those in the training data
             self.val_df = self.val_df[
-                self.val_df[self.label_column].isin(
-                    self.data_df[self.label_column].unique()
+                self.val_df[self.label_column_name].isin(
+                    self.data_df[self.label_column_name].unique()
                 )
             ]
 
         print(f"Training data size: {self.data_df.shape}")
         if self.val_df is not None:
             print(f"Validation data size: {self.val_df.shape}")
 
-        self.class_names = set(self.data_df[self.label_column].values)
+        self.class_names = set(self.data_df[self.label_column_name].values)
         self.label2int = {label: i for i, label in enumerate(self.class_names)}
         self.int2label = {value: key for key, value in self.label2int.items()}
 
-        genes = self.gene_annotation_tdb.df[:]["cellarr_gene_index"].tolist()
+        genes = (
+            self.gene_annotation_tdb.query(attrs=["cellarr_gene_index"])
+            .df[:]["cellarr_gene_index"]
+            .tolist()
+        )
```

Comment:
@tony-kuo this is faster than accessing the entire dataframe from tiledb

Reply:
Nice, didn't know you could query with no condition to grab a column.